src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileGlobals.java

```diff
@@ -637,6 +644,56 @@ public void useRepo(
     }
   }
 
+  @StarlarkMethod(
+      name = "override_repo",
+      doc =
+          "Makes one or more repos in the scope of the current module visible in the scope of the"
+              + " given module extension under the specified names. This can be used both to make"
+              + " new repos visible to other repos generated by the extension and to override repos"
+              + " generated by the extension.",
+      parameters = {
+        @Param(
+            name = "extension_proxy",
+            doc = "A module extension proxy object returned by a <code>use_extension</code> call."),
+      },
+      extraPositionals =
+          @Param(
+              name = "args",
+              doc =
+                  "The names of the repos to make visible to the other repositories generated by"
+                      + " the extension."),
+      extraKeywords =
+          @Param(
+              name = "kwargs",
+              doc =
+                  "The names of the repos to make visible to the other repositories generated by"
+                      + " the extension, where the values are the names of repos in the scope of"
+                      + " the current module and the keys are the names under which they will be"
+                      + " visible in the scope of the extension."),
+      useStarlarkThread = true)
+  public void overrideRepo(
+      ModuleExtensionProxy extensionProxy,
+      Tuple args,
+      Dict<String, Object> kwargs,
+      StarlarkThread thread)
+      throws EvalException {
+    ModuleThreadContext context = ModuleThreadContext.fromOrFail(thread, "override_repo()");
+    context.setNonModuleCalled();
+    if (context.shouldIgnoreDevDeps()) {
+      // Ignore calls early as they may refer to repos that are dev dependencies (or this is not the
+      // root module).
+      return;
+    }
+    ImmutableList<StarlarkThread.CallStackEntry> stack = thread.getCallStack();
```

Comment:
why do we need the call stack? everything in MODULE.bazel files should be at the "top" layer, so `thread.getCallerLocation()` should suffice, no?

Reply:
That wouldn't give us stack traces and thus source file snippets. Inconsistent Starlark error reporting is a pet peeve of mine. I think that consistently showing stack traces with source snippets is better than sometimes just showing locations, even if the stack traces only have one frame.

Reply:
for my curiosity, do you have a paste handy of the stack traces with source snippets? How does it work with multiple frames actually -- does each frame get a line shown, a la Python?

If we're using a (single-frame) stack here, we should ideally use it everywhere else, too. (by that I mean `use_repo` etc.) But also why do we store just a `Location` in `ModuleExtensionUsage.RepoOverride`? Maybe those should store stacks too.

Reply:
> for my curiosity, do you have a paste handy of the stack traces with source snippets? How does it work with multiple frames actually -- does each frame get a line shown, a la Python?

Never mind this, an example is in the tests (and indeed it's like Python). The other question still stands.

Reply:
We just store a `Location` since we can't show two stacks for one error. The error will be on the `override_repo` call, but at least it references the import that makes the overriding repo overridden.

And yes, I agree that we should use it in more places, but I would like to do that in a follow-up.

Reply:
OK, that's fair.